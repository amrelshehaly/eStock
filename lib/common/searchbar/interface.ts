export default interface Props {
  setSearch: (search: string) => void
  onSubmit: (e: any) => void
  search: string
}
