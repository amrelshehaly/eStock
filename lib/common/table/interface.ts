export interface HeaderRow {
    id?: string
    key: string[]
    type?: string
    name: string
    format?: (cell: unknown) => string 
  }

export interface DataRow {
    headers: HeaderRow[]
    data: Record<string, unknown>[]
    actions?: React.ReactNode
  }