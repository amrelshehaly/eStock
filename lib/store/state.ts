import { derived } from 'overmind'
import { base } from '@lib/models/base.interface'
import { Stock } from '@lib/models/stock.interface'
import { StockDetails, PreviousClose } from '@lib/models/stockdetails.interface'

type Props = {
  base: base
  stock: Stock
  stockDetails: StockDetails
  previousClose: PreviousClose
}

export const state: Props = {
  base: {
    count: 0,
    currentPage: 0,
    error: '',
    index: {
      start: 0,
      end: 16,
    },
    loading: false,
    memory: [],
    page: 'Home',
    search: '',
    startSearching: false,
  },
  stock: {
    next_url: '',
    results: [],
  },
  stockDetails: {
    description: '',
    name: '',
    branding: {
      icon_url: '',
      logo_url: '',
    },
    sic_description: '',
    ticker: '',
    homepage_url: '',
  },
  previousClose: {
    c: 0,
    l: 0,
    h: 0,
    o: 0,
  },
}

// export interface ErrorHandling {
//   success: boolean;
//   error?: { message: string };
// }
