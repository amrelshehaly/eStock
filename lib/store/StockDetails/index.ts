import { StockDetailsState } from "./state";
import * as actions from "./actions";
import * as effects from "./effects";

export { StockDetailsState, actions, effects };
