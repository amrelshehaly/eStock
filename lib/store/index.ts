import { IContext } from "overmind";
import { merge } from "overmind/config";
import {
  createActionsHook,
  createEffectsHook,
  createReactionHook,
  createStateHook,
} from "overmind-react";
import * as base from "./base";
import * as StockDetails from "./StockDetails"

export const storeConfig = merge(base, StockDetails,{});

export type IAppContext = IContext<typeof storeConfig>;

export const useAppState = createStateHook<IAppContext>();
export const useActions = createActionsHook<IAppContext>();
export const useEffects = createEffectsHook<IAppContext>();
export const useReaction = createReactionHook<IAppContext>();
