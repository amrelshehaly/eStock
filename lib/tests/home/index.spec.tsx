import React from 'react'
import { createOvermindMock } from 'overmind'
import { storeConfig } from '../../store/index'
import { promises } from 'stream'
import { Stock } from '@lib/models/stock.interface'

describe('base state', () => {
  const overmind = createOvermindMock(storeConfig, (state) => {
    state.base.error = 'This is an error msg'
  })
  it('check if reseting error msg', async () => {
    await overmind.actions.base.ResetErrorMsg()
    expect(overmind.state.base.error).toEqual('')
  })
})

describe('calling an api', () => {
  it('should get tickers list from the api', async () => {
    const overmind = createOvermindMock(storeConfig, {
      Stock: {
        api: {
          getTickers: async (): Promise<Stock> => {
            return await Promise.resolve({
              next_url: '',
              results: [
                {
                  name: 'AAPL',
                  ticker: 'Apple Inc.',
                },
              ],
            })
          },
        },
      },
    })

    await overmind.actions.base.onInitializeOvermind()

    expect(overmind.state.stock.results).toEqual([
      {
        name: 'AAPL',
        ticker: 'Apple Inc.',
      },
    ])
  })
})

describe('Searching for a ticker', () => {
  it('should have a an array with the ticker name if found', async () => {
    const overmind = createOvermindMock(
      storeConfig,
      {
        Stock: {
          api: {
            searchForTicker: async (ticker: string): Promise<Stock> => {
              return await Promise.resolve({
                next_url: '',
                results: [
                  {
                    ticker: 'AAPL',
                    name: 'Apple Inc.',
                  },
                ],
              })
            },
          },
        },
      },
      (state) => {
        state.base.search = 'AAPL'
      },
    )

    await overmind.actions.Stock.SearchForStock()

    expect(overmind.state.stock.results).toContainEqual(
      {
        ticker: 'AAPL',
        name: 'Apple Inc.',
      },
    )
  })
})


