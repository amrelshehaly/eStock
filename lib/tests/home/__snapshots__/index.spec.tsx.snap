// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling an api should get tickers list from the api 1`] = `[]`;
